man/sd_event_source_set_floating.xml

```diff
@@ -0,0 +1,120 @@
+
+  <refentryinfo>
+    <title>sd_event_source_set_floating</title>
+    <productname>systemd</productname>
+  </refentryinfo>
+
+  <refmeta>
+    <refentrytitle>sd_event_source_set_floating</refentrytitle>
+    <manvolnum>3</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>sd_event_source_set_floating</refname>
+    <refname>sd_event_source_get_floating</refname>
+
+    <refpurpose>Set or retrieve 'floating' state of event sources</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <funcsynopsis>
+      <funcsynopsisinfo>#include &lt;systemd/sd-event.h&gt;</funcsynopsisinfo>
+
+      <funcprototype>
+        <funcdef>int <function>sd_event_source_set_floating</function></funcdef>
+        <paramdef>sd_event_source *<parameter>source</parameter></paramdef>
+        <paramdef>int <parameter>floating</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>int <function>sd_event_source_get_floating</function></funcdef>
+        <paramdef>sd_event_source *<parameter>source</parameter></paramdef>
+      </funcprototype>
+
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_event_source_set_floating()</function> may be used to control whether the reference
+    counter of the event source object specified as <parameter>source</parameter> is automatically
+    decremented when the event loop object it is associated with is freed. The various calls that allocate
+    event source objects (i.e.
+    <citerefentry><refentrytitle>sd_event_add_io</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
+    <citerefentry><refentrytitle>sd_event_add_time</refentrytitle><manvolnum>3</manvolnum></citerefentry> and
+    similar) will automatically set an event source object to 'floating' mode if the caller passed
+    <constant>NULL</constant> in the parameter used to return a reference to the event source object. With
+    <function>sd_event_source_get_floating()</function> the 'floating' state can be changed after
+    creation. Generally, it is useful to set event sources to 'floating' that shall live as long as the event
+    loop they are associated with lives and where automatic rather than explicit clean-up is preferable. In
+    simple cases, a 'floating' event source object is easily created simply by passing
+    <constant>NULL</constant> to the return parameter of the call that creates the event source object;
+    however in some cases where the 'floating' concept would be useful it's necessary to adjust additional
+    event source properties after allocation (e.g. its priority or description string). In those cases it is
+    possible to acquire a regular reference to the object first (i.e. create it 'non-floating'), so that
+    these properties can be changed, and only mark it as 'floating' afterwards. The
+    <parameter>floating</parameter> parameter is a boolean: zero to disable 'floating' mode, non-zero to
+    enable it.</para>
```

Comment:
I know what floating does, so I understand what this is saying, but if I didn't, I would be completely baffled. It's not incorrect, but still.

Maybe something like this:
> sd_event_source_set_floating() takes a boolean and sets the "floating" state of the <parameter>source</parameter> object. This is used to change the direction of reference counts for the object and the event loop it is associated with. In non-floating mode, the source object holds a reference to the event loop object, but not vice versa. The creator of the source object must hold a reference to it as long as the source should exist. In floating mode, the event loop holds a reference to the source object, and will decrease the reference count when being freed. This means that a reference to the event loop should be held to prevent both from being destroyed.
>
> Various calls that allocate event source objects (i.e. <citerefentry><refentrytitle>sd_event_add_io</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
>    <citerefentry><refentrytitle>sd_event_add_time</refentrytitle><manvolnum>3</manvolnum></citerefentry> and
>    similar) will automatically set an event source object to 'floating' mode if the caller passed
>    <constant>NULL</constant> in the parameter used to return a reference to the event source object.
> Nevertheless, it may be necessary to gain temporary access to the source object, for example to adjust  event source properties after allocation (e.g. its priority or description string). In those cases the object may be created in non-floating mode, and the returned reference used to adjust the properties, and the object marked as floating afterwards, and the reference in the caller dropped.